### PR TITLE
modules: mcuboot: Disable Zephyr's MCUboot handling for child images

### PR DESCRIPTION
**Pull request by MarekPieta**
Change disables Zephyr's MCUboot handling in child images. The NCS handles needed operations on its own. Change is needed to ensure consistency for main application image and child images.

Jira: NCSDK-22347

### CODE REVIEW

**Review by de-nordic — APPROVED** [review]

**Comment by de-nordic** [comment]
@nordicjm Can you take a look?

**Comment by nordicjm** [comment]
I do not think this is right and if the idea is to propagate up to the target ncs image, this will not work with sysbuild-ncs @tejlmand

**Comment by MarekPieta** [comment]
Rebased

**Comment by MarekPieta** [comment]
> I do not think this is right and if the idea is to propagate up to the target ncs image, this will not work with sysbuild-ncs @tejlmand

It seems that we cannot remove the `PARENT_SCOPE` here (the information would no longer be propagated to Zephyr build system so it would not work as expected). I also understand that the solution is temporary, has its drawbacks and may need to be improved later (I copied it from the bottom of the `CMakeLists.txt` file to make handling `mcuboot_secondary_app` variant image consistent with the main application image).

**Comment by rlubos** [comment]
@nordicjm @tejlmand Any feedback wrt above comment?

File: modules/mcuboot/CMakeLists.txt [hunk 1]
```diff
@@ -28,6 +28,10 @@ endif()
 
 if(CONFIG_BOOTLOADER_MCUBOOT)
   if (CONFIG_NCS_IS_VARIANT_IMAGE)
+    # NCS Handles everything regarding mcuboot, ensure Zephyr doesn't interfere.
+    # This is a temporary solution until Zephyr signing has been made more modular.
+    set(CONFIG_BOOTLOADER_MCUBOOT False PARENT_SCOPE)
```
Comment (tejlmand) [inline modules/mcuboot/CMakeLists.txt:33]: this is not safe.

CONFIG_BOOTLOADER_MCUBOOT is a setting imported from the `.config`.
I acknowledge that it only impacts CMake configure time, and that CONFIG_BOOTLOADER_MCUBOOT is still set in autoconf.h and elsewhere in the system, such as python code loading the `.config`.

But in CMake, there can be customer code checking for this flag for other purposes than signing.

Has it been investigated why this happens for XIP images but not for other variant builds ?

**Review by tejlmand — CHANGES_REQUESTED** [review]
We should not start to manipulate Kconfig settings inside CMake.

Can result in unintended / unexpected errors, especially at NCS downstream users who are not aware of such manipulation.

File: modules/mcuboot/CMakeLists.txt (hunk 1 above)
Comment (MarekPieta) [reply modules/mcuboot/CMakeLists.txt:33]: Hi @tejlmand,

Yes, I am aware that the solution is not perfect and we should not manipulate Kconfig values inside CMake. I just make the behavior consistent with handling main application image - I actually used the same workaround (please take a look at the bottom of this CMakeLists.txt file, you will see the same hotfix applied for the main image):
```
# NCS Handles everything regarding mcuboot, ensure Zephyr doesn't interfere.
# This is a temporary solution until Zephyr signing has been made more modular.
set(CONFIG_BOOTLOADER_MCUBOOT False PARENT_SCOPE)
```

Comment (MarekPieta) [reply modules/mcuboot/CMakeLists.txt:33]: Could you provide a proper solution for the problem during the ongoing upmerge? Or maybe we could accept my temporary workaround and improve it later on (together with improving integration for the main application image)?

Comment (tejlmand) [reply modules/mcuboot/CMakeLists.txt:33]: I see that.
Not fond of that solution either, but at least you are consistent with what we are already doing.

Solution accepted, but please do a rebase.

Comment (MarekPieta) [reply modules/mcuboot/CMakeLists.txt:33]: I need to open a new PR to sdk-nrf main branch. Give me moment.

**Comment by MarekPieta** [comment]
Upmerge is done already, opened PR to main that introduces the change: https://github.com/nrfconnect/sdk-nrf/pull/11696.
Closing this one

**Pull request closed without merging**